virtool/samples/api.py

```diff
@@ -707,16 +704,15 @@ async def upload_cache_reads(req):
     name = req.match_info["filename"]
     sample_id = req.match_info["sample_id"]
     key = req.match_info["key"]
+    config = get_config_from_req(req)
 
     if name not in ["reads_1.fq.gz", "reads_2.fq.gz"]:
         raise HTTPBadRequest(text="File name is not an accepted reads file")
 
-    caches_path = req.app["config"].data_path / "caches"
+    caches_path = join_cache_path(config, key)
     await asyncio.to_thread(caches_path.mkdir, parents=True, exist_ok=True)
 
-    cache_path = (
-        Path(virtool.caches.utils.join_cache_path(req.app["config"], key)) / name
-    )
+    cache_path = caches_path / name
```

Comment:
Naming is not great here either.

This path is for a file (`name` is the filename) in the cache not the cache itself as suggested by the name `cache_path`. Change this to something like `cache_file_path`.